samples/notebooks/DSL Static Type Checking.ipynb

```diff
@@ -0,0 +1,732 @@
+   "metadata": {},
+   "source": [
+    "# Type Check with YAML components: failed scenario"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Author components in YAML"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "component_a = '''\\\n",
+    "name: component a\n",
+    "description: component a desc\n",
+    "inputs:\n",
+    "  - {name: field_l, type: Integer}\n",
+    "outputs:\n",
+    "  - {name: field_m, type: {GCSPath: {openapi_schema_validator: {type: string, pattern: \"^gs://.*$\" } }}}\n",
+    "  - {name: field_n, type: customized_type}\n",
+    "  - {name: field_o, type: GcrUri} \n",
+    "implementation:\n",
+    "  container:\n",
+    "    image: gcr.io/ml-pipeline/component-a\n",
+    "    command: [python3, /pipelines/component/src/train.py]\n",
+    "    args: [\n",
+    "      --field-l, {inputValue: field_l},\n",
+    "    ]\n",
+    "    fileOutputs: \n",
+    "      field_m: /schema.txt\n",
+    "      field_n: /feature.txt\n",
+    "      field_o: /output.txt\n",
+    "'''\n",
+    "component_b = '''\\\n",
+    "name: component b\n",
+    "description: component b desc\n",
+    "inputs:\n",
+    "  - {name: field_x, type: customized_type}\n",
+    "  - {name: field_y, type: GcsUri}\n",
+    "  - {name: field_z, type: {GCSPath: {openapi_schema_validator: {type: string, pattern: \"^gs://.*$\" } }}}\n",
+    "outputs:\n",
+    "  - {name: output_model_uri, type: GcsUri}\n",
+    "implementation:\n",
+    "  container:\n",
+    "    image: gcr.io/ml-pipeline/component-a\n",
+    "    command: [python3]\n",
+    "    args: [\n",
+    "      --field-x, {inputValue: field_x},\n",
+    "      --field-y, {inputValue: field_y},\n",
+    "      --field-z, {inputValue: field_z},\n",
+    "    ]\n",
+    "    fileOutputs: \n",
+    "      output_model_uri: /schema.txt\n",
+    "'''"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Author a pipeline with the above components"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "type name GcrUri is different from expected: GcsUri\n",
+      "Component \"component b\" is expecting field_y to be type(GcsUri), but the passed argument is type(GcrUri)\n"
+     ]
+    }
+   ],
+   "source": [
+    "import kfp.components as comp\n",
+    "import kfp.dsl as dsl\n",
+    "import kfp.compiler as compiler\n",
+    "from kfp.dsl._types import InconsistentTypeException\n",
+    "task_factory_a = comp.load_component_from_text(text=component_a)\n",
+    "task_factory_b = comp.load_component_from_text(text=component_b)\n",
+    "\n",
+    "#Use the component as part of the pipeline\n",
+    "@dsl.pipeline(name='type_check_b',\n",
+    "    description='')\n",
+    "def pipeline_b():\n",
```

Comment:
Adding more inline comments to explain how the type checking works in the sample code.

Reply:
Done.